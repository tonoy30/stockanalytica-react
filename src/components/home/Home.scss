.home {
	button {
		margin: 10px;
		padding: 0.5rem 1rem;
		cursor: pointer;
		&:hover {
			background-color: #cacdf1;
			border-radius: 5px;
			transition: all 0.2s ease-in-out;
		}
	}
}
