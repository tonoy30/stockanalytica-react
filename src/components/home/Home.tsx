import React from 'react'
import { useDispatch } from "react-redux"
import { defaultAction } from '../../actions/default.action'
import "./Home.scss"

const Home = () => {
	const dispatch = useDispatch()
	return (
		<div className="home">
			<div>
				Lorem ipsum dolor sit amet consectetur adipisicing elit. Dicta, dignissimos.
			</div>
			<button onClick={() => dispatch(defaultAction())}>Dispatch</button>
		</div>
	)
}

export default Home
