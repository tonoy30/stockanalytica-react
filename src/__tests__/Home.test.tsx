import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../components/home/Home';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { Store, AnyAction } from 'redux';
import { defaultAction } from '../actions/default.action';
const mockStore = configureStore([]);

describe('Home Test', () => {
	let store: Store<any, AnyAction>;
	let component: renderer.ReactTestRenderer;

	beforeEach(() => {
		store = mockStore({});
		store.dispatch = jest.fn();
		component = renderer.create(
			<Provider store={store}>
				<Home />
			</Provider>
		);
	});
	it('should render with given state from Redux store', () => {
		expect(component.toJSON()).toMatchSnapshot();
	});
	it('should dispatch an action on button click', () => {
		renderer.act(() => {
			component.root.findByType('button').props.onClick();
		});

		expect(store.dispatch).toHaveBeenCalledTimes(1);
		expect(store.dispatch).toHaveBeenCalledWith(
			defaultAction()
		);
	});
});
