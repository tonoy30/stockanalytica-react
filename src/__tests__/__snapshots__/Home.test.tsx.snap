// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Test should render with given state from Redux store 1`] = `
<div>
  Home
  <button
    onClick={[Function]}
  >
    Dispatch
  </button>
</div>
`;
